pyxem/utils/_deprecated.py

```diff
@@ -0,0 +1,134 @@
+"""Helper functions and classes for managing kikuchipy.
+This module and documentation is only relevant for kikuchipy developers,
+not for users.
+.. warning:
+    This module and its submodules are for internal use only.  Do not
+    use them in your own code. We may change the API at any time with no
+    warning.
+"""
+
+import functools
+import inspect
+from typing import Callable, Optional, Union
+import warnings
+
+import numpy as np
+
+
+class deprecated:
+    """Decorator to mark deprecated functions with an informative
+    warning.
+    Adapted from
+    `scikit-image
+    <https://github.com/scikit-image/scikit-image/blob/main/skimage/_shared/utils.py#L297>`_
```

Comment:
Which line has `class deprecated` changed, so I suggest to drop that.

```suggestion
    <https://github.com/scikit-image/scikit-image/blob/main/skimage/_shared/utils.py>`_
```